src/wp-includes/class-wp-taxonomy.php

```diff
@@ -510,7 +510,12 @@ public function add_rewrite_rules() {
 				$tag = '([^/]+)';
 			}
 
-			add_rewrite_tag( "%$this->name%", $tag, $this->query_var ? "{$this->query_var}=" : "taxonomy=$this->name&term=" );
+			$query = $this->query_var ? "{$this->query_var}=" : "taxonomy=$this->name&term=";
+
+			add_rewrite_tag( "%taxonomy-$this->name%", "$this->name()", $query );
```

Comment:
The trailing `()` here is a RegEx capturing group. WordPress really wants to retrieve a match from this RegEx that it'll use as the value in the query string; it'll automatically build a query string like this: `?key=$matches[0]`, using the `$matches` var resulting from applying the RegEx.

Since we don't want to set any value, we work around this by providing a capturing group that will return the empty string, thus resulting in `?key=`.